Cloud Auth Key Pairs UI

mansam opened the pull request:
Adds Key Pairs subheading under Cloud for the `ManageIQ::Providers::CloudManager::AuthKeyPair` class.

![key_pairs_list](https://cloud.githubusercontent.com/assets/628956/11879409/3464f9ea-a4c9-11e5-92d6-823a4e7d9728.png)
![key_pairs_details](https://cloud.githubusercontent.com/assets/628956/11879407/320c1070-a4c9-11e5-9862-16a55c180944.png)

Fryguy commented:
@dclarizio @blomquisg @Ladas Please review

chessbyte commented:
@mansam Please rebase and push again

martinpovolny commented:
@mansam : can you, please, add some spec for the new UI?

martinpovolny commented on app/helpers/application_helper.rb:
```diff
@@ -765,7 +765,7 @@ def display_back_button?
   end
 
   def display_adv_search?
-    %w(availability_zone cloud_volume container_group container_node container_service
```
You are starting new controller, maybe we your use this opportunity to cleanup this a bit and introduce something like `controller.had_adv_search?` and test for that first. Instead of adding to the ugly list we have here.
Does it make sense? Guess there will not be advanced search for any of this controller's screens, right?

martinpovolny commented on app/controllers/auth_key_pair_cloud_controller.rb:
```diff
@@ -0,0 +1,103 @@
+    @auth_key_pair_cloud = @record = identify_record(params[:id])
+    return if record_no_longer_exists?(@auth_key_pair_cloud)
+
+    @gtl_url = "/auth_key_pair_cloud/show/" << @auth_key_pair_cloud.id.to_s << "?"
```
please, use string interpolation instead

martinpovolny commented on app/controllers/auth_key_pair_cloud_controller.rb:
```diff
@@ -0,0 +1,103 @@
+    )
+
+    case @display
+    when "download_pdf", "main", "summary_only"
```
please, prefer %w(...)

mansam replied:
I might be confused but that doesn't seem to work in a "when" clause?

martinpovolny replied:
You are right, it does not work there.

martinpovolny commented on app/controllers/auth_key_pair_cloud_controller.rb:
```diff
@@ -0,0 +1,103 @@
+      @showtype = @display
+      if @view.extras[:total_count] && @view.extras[:auth_count] &&
+         @view.extras[:total_count] > @view.extras[:auth_count]
+        @bottom_msg = "* You are not authorized to view " +
```
Please, use gettext here. Generally try to make all new code i18n compliant.

martinpovolny commented on app/controllers/auth_key_pair_cloud_controller.rb:
```diff
@@ -0,0 +1,103 @@
+  private
+
+  def get_session_data
+    @title      = "Key Pair"
```
gettext?

mansam commented:
I've added a spec for the controller, and fixed (I believe) the i18n problems. Hopefully I did that right. You're right that auth key pairs don't need adv search so I removed that from the list. Still thinking about how to make that check a bit better.

martinpovolny commented:
@mansam : as with the quotas: I see no way to get back from the details to the list of keypairs. Also the assignet tags are not shown in the keypair detail page. Plus, please, update the toolbars to match the recent changes (icons). Thx!

martinpovolny commented:
Looks good. I think we can merge this.

One last question: Why do we need the change to `app/services/request_referer_service.rb` is the list of keypairs and/or a keypair going to be directly referenced from outside so that a GET request w/o a referer is needed?

mansam commented:
I (mistakenly) thought an entry there was required. I'll remove it.
